packages/block-library/src/site-logo/style.scss

```diff
@@ -18,12 +18,7 @@
 	img {
 		height: auto;
 		max-width: 100%;
-	}
-
-	// Inherit border radius from style variations.
-	a,
-	img {
-		border-radius: inherit;
+		box-sizing: border-box;
```

Comment:
I found that the border radius from the "rounded" style variation is applied without this CSS, but maybe I missed something in my testing.